### PR TITLE
Avoid direct-CXs in RZX calibration pass

### PR DESCRIPTION
**Pull request by nkanazawa1989**
### Summary

RZX calibration pass is implemented for CX with echoed cross resonance sequence. Recent IBM backends started to support another realization of CX with so called direct-CX. Generation of fractional gate for this calibration is not straightforward because dynamics on the target qubit is much more complicated. 

In this PR, passes are updated to avoid termination by errors. This is temporary solution, just not to crash the transpiler. We need some additional research for the technique to stretch the direct-CX pulses. It is expected that these passes are upgraded again with support for direct-CXs.

### Details and comments

In addition to error-avoid mechanism, some cleanup has done in this PR. The logic for error avoidance is very simple. I newly introduced `CalibrationNotAvailable` exception, which is raised when the pass encounters the direct-CX. This is caught by the base class `CalibrationBuilder` to ignore invalid calibrations.

### CODE REVIEW

File: qiskit/transpiler/passes/calibration/pulse_gates.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+from qiskit.pulse import (
+    Schedule,
+    ScheduleBlock,
+)
+from qiskit.pulse.instruction_schedule_map import InstructionScheduleMap
+
+from .base_builder import CalibrationBuilder
+
+
+class PulseGates(CalibrationBuilder):
```
Comment (nkanazawa1989) [inline qiskit/transpiler/passes/calibration/pulse_gates.py:27]: (for review) Just moved to new file. Nothing changed.

File: qiskit/transpiler/passes/calibration/rzx_builders.py [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    GaussianSquare,
+)
+from qiskit.pulse.filters import filter_instructions
+from qiskit.pulse.instruction_schedule_map import InstructionScheduleMap, CalibrationPublisher
+
+from .base_builder import CalibrationBuilder
+from .exceptions import CalibrationNotAvailable
+
+
+class RZXCalibrationBuilder(CalibrationBuilder):
```
Comment (nkanazawa1989) [inline qiskit/transpiler/passes/calibration/rzx_builders.py:39]: (for review) Moved to new file, with some cleanup for `get_calibration` method.

File: qiskit/transpiler/passes/calibration/base_builder.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+from qiskit.circuit import Instruction as CircuitInst
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.pulse.schedule import Schedule, ScheduleBlock
+from qiskit.pulse.instruction_schedule_map import CalibrationPublisher
+from qiskit.transpiler.basepasses import TransformationPass
+
+from .exceptions import CalibrationNotAvailable
+
+
+class CalibrationBuilder(TransformationPass):
```
Comment (nkanazawa1989) [inline qiskit/transpiler/passes/calibration/base_builder.py:27]: (for review) Moved to new file

File: qiskit/transpiler/passes/calibration/base_builder.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        for node in dag.gate_nodes():
+            qubits = [qubit_map[q] for q in node.qargs]
+
+            if self.supported(node.op, qubits) and not dag.has_calibration_for(node):
+                # Calibration can be provided and no user-defined calibration is already provided
+                try:
+                    schedule = self.get_calibration(node.op, qubits)
+                except CalibrationNotAvailable:
+                    # Fail in schedule generation. Just ignore.
+                    continue
```
Comment (nkanazawa1989) [inline qiskit/transpiler/passes/calibration/base_builder.py:73]: (for review) This is updated.

File: releasenotes/notes/upgrade_rzx_builder_skip_direct_cx-d0beff9b2b86ab8d.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    non-ECR sequence.
+deprecations:
+  - |
+    Unused argument ``qubit_channel_mapping`` in
+    :class:`.RZXCalibrationBuilder` and :class:`.RZXCalibrationBuilderNoEcho`
+    transpiler passes has been deprecated will be removed.
+    This argument is no longer used.
+other:
+  - |
+    Transpiler pass module :mod:`~qiskit.transpiler.passes.calibration` has been reorganized.
```
Comment (taalexander) [inline releasenotes/notes/upgrade_rzx_builder_skip_direct_cx-d0beff9b2b86ab8d.yaml:19]:
```suggestion
    The transpiler pass module :mod:`~qiskit.transpiler.passes.calibration` has been reorganized.
```

**Review by taalexander — CHANGES_REQUESTED** [review]
LGTM, just some minor comments.

File: qiskit/transpiler/passes/calibration/builders.py [hunk 6]
```diff
@@ -12,469 +12,9 @@
 
 """Calibration creators."""
 
-from abc import abstractmethod
-from typing import List, Union
+# This pass will be deprecated.
```
[the remark this answers is not in the document]
Comment (nkanazawa1989) [reply qiskit/transpiler/passes/calibration/builders.py]: Yes, deprecation warning will be added in next release and removed afterwards. TODO comment is written as a reminder.

**Review by taalexander — APPROVED** [review]

**Pull request merged**